make 编译不过

编译出错， 环境：ubuntu LTS 16.04

Using built-in specs.
COLLECT_GCC=g++
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/5/lto-wrapper
Target: x86_64-linux-gnu
gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.5)




g++ -O3 -lbz2 -lz   -c -o libHDiffPatch/HDiff/diff.o libHDiffPatch/HDiff/diff.cpp
In file included from libHDiffPatch/HDiff/diff.cpp:29:0:
libHDiffPatch/HDiff/diff.h:63:18: error: expected identifier before ‘*’ token
         size_t  (*maxCompressedSize)(const hdiff_TCompress* compressPlugin,size_t dataSize);
                  ^
libHDiffPatch/HDiff/diff.h:63:76: error: ‘size_t’ has not been declared
         size_t  (*maxCompressedSize)(const hdiff_TCompress* compressPlugin,size_t dataSize);
                                                                            ^
libHDiffPatch/HDiff/diff.h:63:91: error: ISO C++ forbids declaration of ‘size_t’ with no type [-fpermissive]
         size_t  (*maxCompressedSize)(const hdiff_TCompress* compressPlugin,size_t dataSize);
                                                                                           ^
libHDiffPatch/HDiff/diff.h:63:91: error: ‘size_t’ declared as function returning a function
libHDiffPatch/HDiff/diff.h:65:27: error: expected identifier before ‘*’ token
         size_t           (*compress)(const hdiff_TCompress* compressPlugin,
                           ^
libHDiffPatch/HDiff/diff.h:67:93: error: ISO C++ forbids declaration of ‘size_t’ with no type [-fpermissive]
                                      const unsigned char* data,const unsigned char* data_end);
                                                                                             ^
libHDiffPatch/HDiff/diff.h:67:93: error: ‘size_t’ declared as function returning a function
libHDiffPatch/HDiff/diff.h:132:36: error: ‘size_t’ has not been declared
                                    size_t kMatchBlockSize=kMatchBlockSize_default);
                                    ^
libHDiffPatch/HDiff/diff.cpp: In function ‘void {anonymous}::do_compress(std::vector<unsigned char>&, const std::vector<unsigned char>&, const hdiff_TCompress*)’:
libHDiffPatch/HDiff/diff.cpp:423:44: error: ‘const hdiff_TCompress {aka const struct hdiff_TCompress}’ has no member named ‘maxCompressedSize’
         size_t maxCodeSize=compressPlugin->maxCompressedSize(compressPlugin,data.size());
                                            ^
libHDiffPatch/HDiff/diff.cpp:428:41: error: ‘const hdiff_TCompress {aka const struct hdiff_TCompress}’ has no member named ‘compress’
         size_t codeSize=compressPlugin->compress(compressPlugin,
                                         ^
<builtin>: recipe for target 'libHDiffPatch/HDiff/diff.o' failed
make: *** [libHDiffPatch/HDiff/diff.o] Error 1

--- a/README.md
+++ b/README.md
@@ -1,6 +1,6 @@
 **HDiffPatch**
 ================
-Version 2.2.2   
+Version 2.2.3   
 byte data Diff & Patch  C\C++ library.  
 
 ---

--- a/builds/codeblocks/HDiffZ.cbp
+++ b/builds/codeblocks/HDiffZ.cbp
@@ -14,6 +14,10 @@
 				<Compiler>
 					<Add option="-g" />
 				</Compiler>
+				<Linker>
+					<Add library="libbz2" />
+					<Add library="libz" />
+				</Linker>
 			</Target>
 			<Target title="Release">
 				<Option output="bin/Release/HDiffZ" prefix_auto="1" extension_auto="1" />

--- a/builds/codeblocks/HPatchZ.cbp
+++ b/builds/codeblocks/HPatchZ.cbp
@@ -14,6 +14,10 @@
 				<Compiler>
 					<Add option="-g" />
 				</Compiler>
+				<Linker>
+					<Add library="libbz2" />
+					<Add library="libz" />
+				</Linker>
 			</Target>
 			<Target title="Release">
 				<Option output="bin/Release/HPatchZ" prefix_auto="1" extension_auto="1" />

--- a/libHDiffPatch/HDiff/diff.h
+++ b/libHDiffPatch/HDiff/diff.h
@@ -124,7 +124,7 @@ typedef struct hdiff_TStreamCompress{
 //    if increase kMatchBlockSize then run faster and require less memory, but out_diff size increase
 //  NOTICE: out_diff->write()'s writeToPos may be back to update headData!
 //  throw std::runtime_error when I/O error,etc.
-static const int kMatchBlockSize_default = (1<<7);
+static const size_t kMatchBlockSize_default = (1<<7);
 void create_compressed_diff_stream(const hpatch_TStreamInput* newData,
                                    const hpatch_TStreamInput* oldData,
                                    hpatch_TStreamOutput*      out_diff,

--- a/libHDiffPatch/HPatch/patch_types.h
+++ b/libHDiffPatch/HPatch/patch_types.h
@@ -28,6 +28,9 @@
 
 #ifndef HPatch_patch_types_h
 #define HPatch_patch_types_h
+
+#include <string.h>//for size_t
+
 #ifdef __cplusplus
 extern "C" {
 #endif

--- a/unit_test.cpp
+++ b/unit_test.cpp
@@ -33,6 +33,7 @@
 #include <time.h>
 #include <stdio.h>
 #include <string.h>
+#include <stddef.h>//for ptrdiff_t
 #include <stdlib.h>
 #include "math.h"
 #include "libHDiffPatch/HDiff/diff.h"

